src/openrct2/world/Map.cpp

```diff
@@ -301,7 +301,14 @@ int32_t TileElementIteratorNext(TileElementIterator* it)
     if (it->element == nullptr)
     {
         it->element = MapGetFirstElementAt(TileCoordsXY{ it->x, it->y });
-        return 1;
+        if (it->element == nullptr)
+        {
+            return 0;
+        }
+        else
+        {
+            return 1;
+        }
```

Comment:
I'd use a ternary, but I understand some people don't like it (applies elsewhere)